client-spark/common/src/main/java/org/apache/spark/shuffle/celeborn/RunningStageManagerImpl.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.shuffle.celeborn;
+
+import java.util.HashSet;
+
+import org.apache.spark.SparkContext$;
+import org.apache.spark.scheduler.DAGScheduler;
+import org.apache.spark.scheduler.Stage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.celeborn.reflect.DynFields;
+import org.apache.celeborn.spark.RunningStageManager;
+
+public class RunningStageManagerImpl implements RunningStageManager {
+
+  private static final Logger LOG = LoggerFactory.getLogger(RunningStageManagerImpl.class);
+  private static final DynFields.UnboundField idField =
+      DynFields.builder().hiddenImpl(Stage.class, "id").build();
+
+  private HashSet<?> runningStages() {
+    try {
+      DAGScheduler dagScheduler = SparkContext$.MODULE$.getActive().get().dagScheduler();
+      DynFields.UnboundField runningStagesField =
+          DynFields.builder().hiddenImpl(DAGScheduler.class, "runningStages").build();
+      return (HashSet<?>) runningStagesField.bind(dagScheduler).get();
```

Comment:
https://github.com/apache/spark/blob/54eb1a2f863bd7d8706c5c9a568895adb026c78d/core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala#L29

https://github.com/apache/spark/blob/54eb1a2f863bd7d8706c5c9a568895adb026c78d/core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala#L166C46-L166C53

Here the `runningStages` type is `scala.collection.mutable.HashSet`, would it be a problem?

Reply:
good catch, I am pretty surprised that it didn't fail.....

Reply:
for being safe, I converted to the exact type

Reply:
because the code in this class is not covered by UT

Reply:
could you cover it likes SparkUtilsSuite？

Reply:
I added a specific test for this method

but I think this method was actually covered already, I instrumented the method with System.out.println() , I can see those were printed ... still not sure why it didn't fail